spec/requests/v1/term_crud_spec.rb

```diff
@@ -0,0 +1,279 @@
+          term_type: 'temporary',
+        }
+      end
+
+      it 'creates term record' do
+        expect(Term.count).to be 1
+        expect(Term.first.uri).to start_with('temp:')
+        expect(Term.first.pref_label).to eql 'Hippogriff'
+      end
+
+      it 'return newly created term in json' do
+        expect(response.body).to be_json_eql(%(
+          {
+            "pref_label": "Hippogriff",
+            "alt_label": [],
+            "authority": null,
+            "term_type": "temporary",
+            "classification": null
+          }
+        )).excluding('uri', 'uuid')
+      end
+
+      it 'returns 201' do
+        expect(response.status).to be 201
+      end
+    end
+
+    context 'when uri is missing for external term' do
+      before do
+        post '/api/v1/vocabularies/mythical_creatures/terms', params: {
+          pref_label: 'Minotaur (Greek mythological character)',
+          authority: 'fast',
+          term_type: 'external',
+          harry_potter_reference: false
+        }
+      end
+
+      it 'returns 400' do
+        expect(response.status).to be 400
+      end
+
+      it 'returns error in json' do
+        expect(response.body).to be_json_eql(%(
+          { "errors": [
+            { "title": "Uri is not valid" },
+            { "title": "Uri can't be blank" },
+            { "title": "Uri hash can't be blank" }
+          ]
+          }
+        ))
+      end
+    end
+  end
+
+  describe 'PATCH /api/v1/vocabularies/:string_key/terms/:uri' do
+    context 'when updating alt_label' do
+      let(:term) { FactoryBot.create(:external_term, vocabulary: vocabulary) }
+
+      before do
+        patch "/api/v1/vocabularies/mythical_creatures/terms/#{CGI.escape(term.uri)}", params: {
+          alt_label: ['Uni']
+        }
+      end
+
+      it 'updates alt_labels for term' do
+        term.reload
+        expect(term.alt_label).to contain_exactly 'Uni'
+      end
+
+      it 'returns updated term' do
+        expect(response.body).to be_json_eql(%(
+          {
+            "uri": "http://id.worldcat.org/fast/1161301/",
+            "pref_label": "Unicorns",
+            "alt_label": ["Uni"],
+            "authority": "fast",
+            "term_type": "external",
+            "classification": null
+          }
+        )).excluding('uuid')
+      end
+
+      it 'returns 200' do
+        expect(response.status).to be 200
+      end
+    end
+
+    context 'when updating term_type' do
+      let(:term) { FactoryBot.create(:external_term, vocabulary: vocabulary) }
+
+      before do
+        patch "/api/v1/vocabularies/mythical_creatures/terms/#{CGI.escape(term.uri)}", params: {
+          term_type: 'local'
+        }
+      end
+
+      it 'does not update record' do
+        term.reload
+        expect(term.uri).not_to eql 'local'
+      end
+
+      it 'returns 200' do
+        expect(response.status).to be 200
+      end
+    end
+
+    context 'when invalid uri' do
```

Comment:
```suggestion
    context 'when attempting to update a uri that is not associated with a known term' do
```
I wanted to change this just because the URI itself isn't an invalid URI.